### ISSUE TITLE
Tournaments

### ISSUE THREAD

**Issue by gushogg-blake** [jsonchess/lightsquare#110]
This is the next major feature.  I'm open to any suggestions about how they should be implemented and what kind of tournaments should be supported.

**Comment by antiproton** [comment]
The installation make of tournaments is very complicated. There are very many permanent visitor needs. So the thing really works. An administrator is required to organize tournaments.
Best Regards R. U.

**Comment by gushogg-blake** [comment]
Yeah I see your point here, it does need players...

I would prefer to implement user profiles as static pages, as it doesn't seem right to do it like the rest of the application where the UI sends websocket requests for everything.  I think as the app gets bigger there should be separate parts of it, some with websockets for quick updates (games etc) and some with templates rendered on the server (user profiles).  So that is probably the next step - re-structuring things slightly to allow for a better design.

**Comment by lexisvar** [comment]
Yeah. But I think that the tournament functionality is very important.

**Comment by gushogg-blake** [comment]
I will start writing tournaments then, as I think that they will also be more fun to write.

I'll start off with knockout tournaments and then implement more complex systems once the framework is in place.

**Comment by antiproton** [comment]
I understand. You're good.
R. U.
All this is not trivial :-)

**Comment by gushogg-blake** [comment]
Just a quick update - I am currently working on basic no-time-limit, single-elimination tournaments.  Most of the work so far has been on the server (http://github.com/jsonchess/lightsquared) so check out the latest commits over there if you want.  I will also create an issue for the server and start referencing commits to it.  I expect to have a working prototype up in the next few days.  Server issue: https://github.com/jsonchess/lightsquared/issues/30

**Comment by antiproton** [comment]
Very complicated.
Increment = Gain per Move
Greetings from Berlin R.U.

**Comment by gushogg-blake** [comment]
Update - I am still making progress on this but it will take a bit longer as there are some fairly large refactorings I've decided to undertake so that I can build it onto a clean foundation.  This will benefit the rest of the code as well, by cleaning up existing data sync code and allowing features like this to be added much more easily and quickly in the future.  I am currently working on a data update mechanism like Meteor's [DDP](https://www.meteor.com/ddp) but simpler.  (I looked at using DDP itself but I think it is too tightly dependent on Meteor).

**Comment by antiproton** [comment]
Very pretty!
R.U.
